src/hotspot/share/runtime/mutex.hpp

```diff
@@ -195,6 +200,11 @@ class Mutex : public CHeapObj<mtSynchronizer> {
   const char *name() const                  { return _name; }
 
   void print_on_error(outputStream* st) const;
+
+  // Print all mutexes/monitors that are currently owned by a thread; called
+  // by fatal error handler.
+  static void print_owned_locks_on_error(outputStream* st);
+  static void print_lock_ranks(outputStream* st);
```

Comment:
This splits print_on_error from print_on. Could you move this before print_on_err and make sure that there's a blank line before print_on_error.